internal/services/recoveryservices/site_recovery_replicated_vm_resource.go

```diff
@@ -108,6 +108,19 @@ func resourceSiteRecoveryReplicatedVM() *pluginsdk.Resource {
 				Optional:         true,
 				ValidateFunc:     azure.ValidateResourceID,
 				DiffSuppressFunc: suppress.CaseDifference,
+				ConflictsWith: []string{
+					"target_zone",
+				},
+			},
+			// extracted from commonschema.ZoneSingleOptionalForceNew() as conflictWith is required.
+			"target_zone": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				ConflictsWith: []string{
+					"target_availability_set_id",
+				},
```

Comment:
this isn't needed on both sides - please use `commonschema` here

Reply:
Thanks @tombuildsstuff , @katbyte has helped reverting the change.